tests/unit/test_network_connections.py

```diff
@@ -472,12 +472,20 @@ def test_validate_disallow_none_in_list(self):
                 "list[?]", allow_empty=True
             ),
         )
-        self.assertRaisesRegexp(
-            ValidationError,
-            "must be a string but is 'None'",
-            v.validate,
-            ["pci-0001:00:00.0", None],
-        )
+        if int(sys.version[0]) == 3:
+            self.assertRaisesRegex(
+                ValidationError,
+                "must be a string but is 'None'",
+                v.validate,
+                ["pci-0001:00:00.0", None],
+            )
+        else:
+            self.assertRaisesRegexp(
+                ValidationError,
+                "must be a string but is 'None'",
+                v.validate,
+                ["pci-0001:00:00.0", None],
+            )
```

Comment:
1) This needs to be a helper that can be used instead of using the if condition in each affected test case and duplicating the assertion.

2) Instead of checking the python version, the code should use introspection to check which method is available

3) This seems to be addressed by Python Six, so maybe you can use that one: https://six.readthedocs.io/